lib/generators.nix

```diff
@@ -278,40 +278,54 @@ rec {
         mapAny 0;
 
   /* Pretty print a value, akin to `builtins.trace`.
-    * Should probably be a builtin as well.
-    */
+   * Should probably be a builtin as well.
+   * The pretty-printed string should be suitable for rendering default values
+   * in the NixOS manual. In particular, it should be as close to a valid Nix expression
+   * as possible.
+   */
   toPretty = {
     /* If this option is true, attrsets like { __pretty = fn; val = …; }
        will use fn to convert val to a pretty printed representation.
        (This means fn is type Val -> String.) */
     allowPrettyValues ? false,
     /* If this option is true, the output is indented with newlines for attribute sets and lists */
-    multiline ? true
+    multiline ? true,
+    /* Render paths with these prefixes using angle bracket syntax, e.g.
+         toPretty { searchPath = [ { name = "foo"; prefix = "/path/to/foo"; } ]; } /path/to/foo/bar/qux
+         ==> "<foo/bar/qux>"
+       Type: [ { name = string; prefix = string or path; } ] */
+    searchPath ? []
```

Comment:
I guess *semantically* this should be a mapping from names to paths, but *technically* it should be a mapping from paths to names (because that's the direction of the transformation we're doing), so I went the neutral route and picked neither.

I'm fine with going the semantic route though.